vmdb/spec/lib/miq_automation_engine/miq_ae_state_machine_new_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+require 'timecop'
+
+describe "MiqAeStateMachine" do
+  before do
+    DummyWorkspace = Class.new do
+      def initialize(options = {})
```

Comment:
Technically, you don't need to have workspace_class as it's own let, but It is most similar to your current code.

@eclarizio thoughts - am I making this too complicated?